certbot/util.py

```diff
@@ -278,76 +277,52 @@ def get_filtered_names(all_names):
     return filtered_names
 
 
-def get_os_info(filepath="/etc/os-release"):
+def get_os_info():
     """
     Get OS name and version
 
-    :param str filepath: File path of os-release file
     :returns: (os_name, os_version)
     :rtype: `tuple` of `str`
     """
 
-    if os.path.isfile(filepath):
-        # Systemd os-release parsing might be viable
-        os_name, os_version = get_systemd_os_info(filepath=filepath)
-        if os_name:
-            return os_name, os_version
-
-    # Fallback to platform module
-    return get_python_os_info()
+    return (distro.id(), distro.version(pretty=False))
 
 
-def get_os_info_ua(filepath="/etc/os-release"):
+def get_os_info_ua():
     """
     Get OS name and version string for User Agent
 
-    :param str filepath: File path of os-release file
     :returns: os_ua
     :rtype: `str`
     """
 
-    if os.path.isfile(filepath):
-        os_ua = get_var_from_file("PRETTY_NAME", filepath=filepath)
-        if not os_ua:
-            os_ua = get_var_from_file("NAME", filepath=filepath)
-        if os_ua:
-            return os_ua
+    return distro.version(pretty=True)
 
-    # Fallback
-    return " ".join(get_python_os_info())
-
-
-def get_systemd_os_info(filepath="/etc/os-release"):
+def get_systemd_os_info():
```

Comment:
There are no calls to this function after your refactoring so I think we should get rid of it.

It's probably overkill but since I saw plugins using this module when working on https://github.com/certbot/certbot/issues/5775, I would keep its behavior but add a deprecation warning and test for it like I did for the reverter code in https://github.com/certbot/certbot/pull/7198. We can then create an issue and remove it in a subsequent release.